### PR TITLE
refactor: change implementation required measures

### PR DESCRIPTION
**Pull request by adnasa**
- simplifies the implementation of `getRequiredMeasures`
- changes name of `getRequiredMeasures` to `getMatchingMeasures` since that's what we were doing

### CODE REVIEW

File: __tests__/logMeasures.js [hunk 1]
```diff
@@ -49,13 +49,13 @@ function logToConsole({ port, components }, measures) {
...
   } else if (typeof components !== undefined && Array.isArray(components)) {
-    var stats = getRequiredMeasures(components, measures)
+    var stats = getMeasuresByComponentName(components, measures)
 
     logMeasures(stats)
   }
 }
 
-function getRequiredMeasures(components, measures) {
+function getMeasuresByComponentName(components, measures) {
```
Comment (adnasa) [inline __tests__/logMeasures.js]: I noticed with this change that you actually had duplicate implemetation on the test (?)
why is that?

File: src/npm/hook.js [hunk 2]
```diff
@@ -131,20 +129,13 @@ const send = (data, port) => {
   )
 }
 
-const getRequiredMeasures = (components, measures) => {
-  let requiredMeasures = []
-
-  if (!Array.isArray(components)) {
```
Comment (adnasa) [inline src/npm/hook.js]: There is no need for this, since in L73, we already were doing this checkl

Comment (nitin42) [reply src/npm/hook.js]: Oops! Ya totally 👍

File: __tests__/logMeasures.js (hunk 1 above)
Comment (nitin42) [reply __tests__/logMeasures.js]: In `src/npm/hook.js` we are send them to a server listening at a particular port while in tests, we are treating them as a proxy i.e logging it to the console. Anyway, it does makes sense to remove this redundant implementation but I think its fine. Let me know what you think!

**Comment by nitin42** [comment]
Looks good to me 👍

I'll review it properly this weekend and will merge the changes. Thank you so much for the work. Really appreciate your contributions!

File: src/npm/hook.js [hunk 3]
```diff
@@ -131,20 +129,13 @@ const send = (data, port) => {
...
-  measures.forEach(measure => {
-    if (components.includes(measure.componentName)) {
-      requiredMeasures.push(measure)
-    }
-  })
-
-  return requiredMeasures
-}
+const getMeasuresByComponentName = (componentNames, measures) =>
+  measures.reduce(
```
Comment (adnasa) [inline src/npm/hook.js]: I realised that this is `filter`. update coming (eventually)

Comment (nitin42) [reply src/npm/hook.js]: Nice!

**Comment by nitin42** [comment]
Up until now, everything seems fine to me. Great job! Also, if you're comfortable, would you mind if I added you as a collaborator on this project ? Let me know if you are comfortable knowing the build setup and folder structure of the project. Thanks 😃

File: __tests__/logMeasures.js (hunk 1 above)
Comment (adnasa) [reply __tests__/logMeasures.js]: I would think that the less redundant code, the better to reduce duplicate work and increase coverage. Let's take a look at this another time. Leaving this as is doesn't break anything

**Comment by adnasa** [comment]
hey!
I'm not entirely sure as of this writing how much more time I can put into different projects. Keeping up right now is hard enough as it is. 😓 

At the moment, all I'm sure about is; I can contribute code whenever I can, but I can't commit to more than that.

Sorry for the late reply on this.

**Comment by adnasa** [comment]
that `filter` is now in 2d09632

**Comment by nitin42** [comment]
Who! Thank you so much for the contribution. I'll merge the PR this weekend with some more changes.

**Comment by adnasa** [comment]
🚀

**Comment by nitin42** [comment]
Released [`3.0.6`](https://github.com/nitin42/react-perf-devtool/releases/tag/3.0.6). Thank you!

**Pull request merged**